operators/grpc_operators/common/tensor_proto.cpp

```diff
@@ -0,0 +1,190 @@
+  }
+}
+
+template <typename T>
+void TensorProto::copy_data_to_proto(const nvidia::gxf::Handle<nvidia::gxf::Tensor>& tensor,
+                                     ::holoscan::entity::Tensor& tensor_proto,
+                                     const cudaStream_t cuda_stream) {
+  if ((*tensor).storage_type() == nvidia::gxf::MemoryStorageType::kDevice) {
+    void* in_data_ptr = (*tensor).pointer();
+    size_t data_size = (*tensor).bytes_size();
+    std::vector<T> in_data(data_size);
+    CUDA_TRY(cudaMemcpyAsync(
+        in_data.data(), in_data_ptr, data_size, cudaMemcpyDeviceToHost, cuda_stream));
+    tensor_proto.set_data(in_data.data(), data_size);
+    tensor_proto.set_memory_storage_type(holoscan::entity::Tensor::kDevice);
+  } else {
+    tensor_proto.set_data((*tensor).pointer(), (*tensor).size());
+    tensor_proto.set_memory_storage_type(holoscan::entity::Tensor::kSystem);
+  }
+}
+
+void TensorProto::proto_to_gxf_tensor(
+    nvidia::gxf::Entity& gxf_entity,
+    const google::protobuf::Map<std::string, ::holoscan::entity::Tensor>& tensor_map,
+    nvidia::gxf::Handle<nvidia::gxf::Allocator>& allocator, const cudaStream_t cuda_stream) {
+  for (auto tensor_entry : tensor_map) {
+    const holoscan::entity::Tensor& tensor_proto = tensor_entry.second;
+    auto tensor = gxf_entity.add<nvidia::gxf::Tensor>(tensor_entry.first.c_str());
+    if (!tensor) { throw std::runtime_error("Failed to create tensor"); }
+
+    nvidia::gxf::Shape shape({tensor_proto.dimensions().begin(), tensor_proto.dimensions().end()});
+    switch (tensor_proto.primitive_type()) {
+      case holoscan::entity::Tensor::kUnsigned8:
+        tensor.value()->reshape<uint8_t>(
+            shape, memory_storage_type(tensor_proto.memory_storage_type()), allocator);
+        copy_data_to_tensor<uint8_t>(tensor_proto, tensor.value(), cuda_stream);
+        break;
+      case holoscan::entity::Tensor::kUnsigned16:
+        tensor.value()->reshape<uint16_t>(
+            shape, memory_storage_type(tensor_proto.memory_storage_type()), allocator);
+        copy_data_to_tensor<uint16_t>(tensor_proto, tensor.value(), cuda_stream);
+        break;
+      case holoscan::entity::Tensor::kFloat32:
+        tensor.value()->reshape<float>(
+            shape, memory_storage_type(tensor_proto.memory_storage_type()), allocator);
+        copy_data_to_tensor<float>(tensor_proto, tensor.value(), cuda_stream);
+        break;
+      default:
+        throw std::runtime_error("Unsupported primitive type");
+    }
+  }
+}
+nvidia::gxf::MemoryStorageType TensorProto::memory_storage_type(
+    holoscan::entity::Tensor_MemoryStorageType mem_storage_type) {
+  switch (mem_storage_type) {
+    case holoscan::entity::Tensor::kDevice:
+      return nvidia::gxf::MemoryStorageType::kDevice;
+    case holoscan::entity::Tensor::kHost:
+      return nvidia::gxf::MemoryStorageType::kHost;
+    case holoscan::entity::Tensor::kSystem:
+      return nvidia::gxf::MemoryStorageType::kSystem;
+  }
+}
```

Comment:
When compiling I get this warning
`/workspace/holohub/operators/grpc_operators/common/tensor_proto.cpp: In static member function 'static nvidia::gxf::MemoryStorageType holoscan::ops::TensorProto::memory_storage_type(holoscan::entity::Tensor_MemoryStorageType)':
/workspace/holohub/operators/grpc_operators/common/tensor_proto.cpp:144:1: warning: control reaches end of non-void function [-Wreturn-type]
  144 | }
`